pkg/cloud/amazon/aws_kms.go

```diff
@@ -109,23 +113,42 @@ func MakeAWSKMS(uri string, env cloud.KMSEnv) (cloud.KMS, error) {
 	opts := session.Options{}
 	switch kmsURIParams.auth {
 	case "", cloud.AuthParamSpecified:
-		if kmsURIParams.accessKey == "" {
+		// Credentials can either be specified with an access key and secret
+		// or with AssumeRole.
+		if kmsURIParams.accessKey != "" || kmsURIParams.secret != "" {
+			// User appears to be trying to authenticate with an access key.
+			if kmsURIParams.roleArn != "" {
```

Comment:
After the meeting today, I think we should take the input for the role out of the credentials part and put it as a separate param that works for both implicit and specified auth. The auth params would then specify the credentials for the user that will be doing the role assuming. What do you think?

Reply:
Yeah that makes sense to me, and I think that should hopefully clean up the code a bit